src/js/sync-to-drive.js

```diff
@@ -0,0 +1,255 @@
+
+    for (const drive of drives) {
+        console.log(
+            'Sync-To-Drive server [',
+            myPid,
+            ']: Synching to drive [',
+            drive.path,
+            '] -> [',
+            drive.mountpoint,
+            ']'
+        );
+
+        const cmd = (() => {
+            switch (process.platform) {
+                case 'win32':
+                    return null;
+                case 'linux':
+                    return [
+                        'rsync',
+                        '-a',
+                        '--delete-before',
+                        '-b',
+                        '--backup-dir=' + path.join(drive.mountpoint, 'deleted'),
+                        '--ignore-existing',
+                        dataAbsPath,
+                        path.join(drive.mountpoint, 'sync')
+                    ].join(' ');
+                default:
+                    return null;
+            }
+        })();
+
+        if (!cmd) {
+            console.log('Sync-To-Drive server [', myPid, ']: ERROR: No command for syncing!');
+
+            return;
+        }
+
+        console.log('Sync-To-Drive server [', myPid, ']: Executing command:', cmd);
+
+        try {
+            const spwndCmd = spawn(cmd, {
+                detached: true,
+                shell: true,
+                stdio: 'ignore'
+            });
+            spwndCmd.unref();
+        } catch (err) {
+            console.log('Sync-To-Drive server [', myPid, ']: ERROR! Couldnt start sync!');
+        }
+    }
+};
+
+// https://stackoverflow.com/a/58844917
+const isProcessRunning = (processName) => {
+    const cmd = (() => {
+        switch (process.platform) {
+            case 'win32':
+                return 'tasklist';
+            case 'darwin':
+                return 'ps -ax | grep ' + processName;
+            case 'linux':
+                return 'ps -A';
+            default:
+                return false;
+        }
+    })();
+
+    try {
+        const result = execSync(cmd).toString();
+
+        return result.toLowerCase().indexOf(processName.toLowerCase()) > -1;
+    } catch (error) {
+        return null;
+    }
+};
+
+if (PLATFORM === 'win32') {
+    console.error('Sync-To-Drive server [', myPid, ']: Windows is currently not supported!');
+    process.exit();
+}
+
+if (isProcessRunning('rsync')) {
+    console.log('Sync-To-Drive server [', myPid, ']: WARN: Sync in progress');
+    process.exit();
+}
+
+const phpConfig = getConfigFromPHP();
+
+if (!phpConfig) {
+    process.exit();
+} else if (!phpConfig.synctodrive_enabled) {
+    console.log('Sync-To-Drive server [', myPid, ']: WARN: Sync script was disabled by config! Aborting!');
+    process.exit();
+}
+
+/* PARSE PHOTOBOOTH CONFIG */
+const parsedConfig = parseConfig(phpConfig);
+console.log('Sync-To-Drive server [', myPid, ']: Drive names ', ...parsedConfig.drives);
+
+/* WRITE PROCESS PID FILE */
+const pidFilename = phpConfig.folders.tmp + '/synctodrive_server.pid';
```

Comment:
We could also join paths with `path.join`

Reply:
yes, very much - I like that suggestion. Thank you !